std/algebra/twistededwards/point.go

```diff
@@ -171,3 +175,33 @@ func (p *Point) ScalarMulFixedBase(cs *frontend.ConstraintSystem, x, y interface
 
 	return p
 }
+
+// Neg computes the negative of a point in SNARK coordinates
+func (p *Point) Neg(cs *frontend.ConstraintSystem, p1 *Point, curve EdCurve) *Point {
+
+	var r, xNeg frontend.Variable
+
+	switch curve.ID {
+	case ecc.BN254:
+		r = cs.Constant(bn254fr.Modulus())
+		break
+	case ecc.BLS12_381:
+		r = cs.Constant(bls12381fr.Modulus())
+		break
+	case ecc.BLS12_377:
+		r = cs.Constant(bls12377fr.Modulus())
+		break
+	case ecc.BW6_761:
+		r = cs.Constant(bw6761fr.Modulus())
+		break
+	default:
```

Comment:
need to panic on default case when curve is not supported / implemented yet
For example we just added BLS24-315 curve --

Reply:
that's right. Already add panic.